### PR TITLE
Fix the bug with TxContext and current GasModel in /local

### PR DESCRIPTION
**Pull request by ak3n**
This PR fixes the incorrect usage of current `TxContext`, making it use the rewinding `ParendHeader`. Also fixes the problem with using the current gas model.

### CODE REVIEW

File: test/Chainweb/Test/Pact/RemotePactTest.hs [hunk 1]
```diff
@@ -9,6 +9,8 @@
 {-# LANGUAGE ScopedTypeVariables #-}
 {-# LANGUAGE TypeApplications #-}
 
+{-# OPTIONS_GHC -fno-warn-type-defaults #-}
```
Comment (ak3n) [inline test/Chainweb/Test/Pact/RemotePactTest.hs:12]: Sorry, but I decided it's irrational to spend time to please the GHC here.

Comment (edmundnoble) [reply test/Chainweb/Test/Pact/RemotePactTest.hs:12]: Where even was the issue?

**Review by edmundnoble — APPROVED** [review]
Looks good.

File: test/Chainweb/Test/Pact/RemotePactTest.hs (hunk 1 above)
Comment (ak3n) [reply test/Chainweb/Test/Pact/RemotePactTest.hs]: Ok, adding a type hint worked properly this time.

File: test/Chainweb/Test/Pact/RemotePactTest.hs [hunk 2]
```diff
@@ -363,15 +363,34 @@ localPreflightSimTest iot nio = testCaseSteps "local preflight sim test" $ \step
...
     runLocalPreflightClient sid cenv cmd7 >>= \case
       Left e -> assertFailure $ show e
       Right LocalResultLegacy{} ->
         assertFailure "Preflight /local call produced legacy result"
       Right MetadataValidationFailure{} ->
         assertFailure "Preflight produced an impossible result"
       Right (LocalResultWithWarns cr' ws) -> do
-        -- check the presence of metadata
-        assertBool "Preflight result should have metadata" $ isJust $ _crMetaData cr'
+        currentBlockHeight <- getCurrentBlockHeight v cenv sid
```
Comment (edmundnoble) [inline test/Chainweb/Test/Pact/RemotePactTest.hs:374]: I like that you're using the metadata here.

**Comment by jwiegley** [comment]
>>>>> "EA" == Evgenii Akentev ***@***.***> writes:

EA> Sorry, but I decided it's irrational to spend time to please the GHC here.

Saving time today that may impact future readers’ comprehension for years to
come is not an economical savings. If there is no reasonable way to satisfy
GHC, then you must include a comment as to why that directive is being added
to the code, especially if it’s being used to resolve similar issues anywhere
else.

**Pull request merged**